include/libssh/threads.h

```diff
@@ -24,8 +24,56 @@
 #include <libssh/libssh.h>
 #include <libssh/callbacks.h>
 
+#ifdef HAVE_PTHREAD
+
+#include <pthread.h>
+#define SSH_STATIC_MUTEX(mutex) \
+    static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER
```

Comment:
Please init with:

```
pthread_mutexattr_init(&ma);
pthread_mutexattr_settype(&ma, PTHREAD_MUTEX_ERRORCHECK);
pthread_mutex_init(m, &ma);
```

Reply:
This is not addressed yet. I have to check if I can avoid the static initialization.